### PR TITLE
Build: Update build process for OpenBSD

### PR DESCRIPTION
**Pull request by tuftedocelot**
OpenBSD builds appear to have gone a bit stale so make the changes to allow building. Currently, only i686 builds and runs; x86_64 yields kernel panics

### CODE REVIEW

**Comment by bgianfo** [comment]
@tuftedocelot these changes look good, hopefully biggie bot didn't scare you away 😁

**Comment by tuftedocelot** [comment]
hey @bgianfo, ha, no, I just saw the test failure in the local sockets which was something I had changed. I thought I'd better dig more into that and then come back :) Closing the PR might've been a hasty move

File: Userland/Libraries/LibCore/LocalSocket.cpp [hunk 1]
```diff
@@ -74,6 +79,10 @@ pid_t LocalSocket::peer_pid() const
     if (getsockopt(fd(), SOL_LOCAL, LOCAL_PEERPID, &pid, &pid_size) < 0) {
 #elif defined(__FreeBSD__)
     if (getsockopt(fd(), SOL_LOCAL, LOCAL_PEERCRED, &creds, &creds_size) < 0) {
+#elif defined(__OpenBSD__)
+    int error;
+    error = getsockopt(fd(), SOL_SOCKET, SO_PEERCRED, &creds, &creds_size);
```
Comment (bgianfo) [inline Userland/Libraries/LibCore/LocalSocket.cpp:84]: Isn't this identical to the case we already have in the in the `#else` block below?

File: Userland/Libraries/LibCore/System.cpp [hunk 2]
```diff
@@ -118,6 +118,8 @@ ErrorOr<void> sigaction(int signal, struct sigaction const* action, struct sigac
 
 #ifdef __APPLE__
 ErrorOr<sig_t> signal(int signal, sig_t handler)
+#elif defined(__OpenBSD__)
```
Comment (bgianfo) [inline Userland/Libraries/LibCore/System.cpp:121]: We should avoid re-declaring the same signature as above.. how about instead we just modify the above `#ifdef` to something like: 

```cpp
#if defined(__APPLE__) || defined(__OpenBSD__)
```

Comment (bgianfo) [reply Userland/Libraries/LibCore/System.cpp:121]: Same comment for the header...

**Pull request closed without merging**